provisioning_client/devdoc/azure_provisioning_faq.md

```diff
@@ -0,0 +1,39 @@
+# Azure Iot Provisioning Client FAQ
```

Comment:
I want to see more!  It'd be great to have an architecture diagram to see how the two clients work together.  It'd also be great to have documentation around how the provisioning client and device client "should" interact.

Reply:
Yes, we can do that.  Let's have a quick chat to talk about exactly what we do here.

---
In reply to: [254831673](https://github.com/Azure/azure-iot-sdk-c/pull/854#discussion_r254831673) [](ancestors = 254831673)